core/worldcat_overrides/request.py

```diff
@@ -0,0 +1,59 @@
+import urllib2
+import urllib
+from exceptions import StopIteration
+
+from worldcat.exceptions import APIKeyError, APIKeyNotSpecifiedError, \
+                                EmptyQueryError, EmptyRecordNumberError, \
+                                InvalidArgumentError, ExtractError
+from worldcat.request.search import SearchAPIRequest
+from worldcat.util.extract import extract_elements
+
+
+# SRURequestV2 class was created to replace the existing SRURequest class in WorldCat that has an incorrect API URL which is causing 
+# breakage with our pull_titles.py script.  This class is a copy of the existing SRURequest class with the exception of the api_url()
+# method which has been updated to use the correct API URL.  This class should be removed once the WorldCat API is fixed.
+# http_get is also overidden to include User-Agent in request to prevent OCLC API from missclassifying our requests as spam.
+class SRURequestV2(SearchAPIRequest):
+    """request.search.NewSRURequest: queries search API using SRU
+    SRURequests should be used when fielded searching is desired.
+    """
+
+    def __init__(self, **kwargs):
```

Comment:
The `__init__`, `__iter__`, `next`, and `subclass_validator` aren't obviously related to this change. Did I miss a connection? It looks like they're very similar to the upstream `OpenSearchRequest` subclass so perhaps we should directly subclass that?

https://github.com/anarchivist/worldcat/blob/a84110c7da0cfb297e2a347b91b3634c4ccb6f0a/worldcat/request/search.py#L73

Reply:
That is correct: Those are unrelated to the change. I just pulled that higher class to give us more control if needed, but I am happy to make that change.

Reply:
Thanks - hopefully we won't have too much code drift here but just in case I'd prefer to have less to maintain.

Reply:
Changes have been implemented and code is again ready for review.